### ISSUE TITLE
Filter certain SQL queries on instrumentation-pg

### ISSUE THREAD

**Issue by brunoluiz** [open-telemetry/opentelemetry-js-contrib#499]
### Is your feature request related to a problem? Please describe
No

### Describe the solution you'd like to see
In certain cases, the backend will do a simple query to check if the Postgres connections are alive. These queries will end-up being sent to the otel collector, but essentially they don't bring any value for tracing analysis.

### Describe alternatives you've considered
There are no alternatives besides implementing some kind of query filtering in it.

**Comment by dyladan** [comment]
When you say "the backend" are you referring to your application or some default behavior of the `pg` module?

**Comment by dyladan** [comment]
What mechanism would you propose for filtering them? A string match or regex might not be a reliable method.

**Comment by brunoluiz** [comment]
> When you say "the backend" are you referring to your application, an ORM, or some default behavior of the pg module?
I've seen many times something similar to the below code to keep a certain pool of connections alive, not allowing the service to go to a degraded state due to disconnected clients. BUT, reviewing this now it might not be exactly a good practice 👀

> What mechanism would you propose for filtering them? A string match or regex might not be a reliable method.
Well, for simple "heartbeat" queries such as below, string match should be fine... But well, for any more involved queries it would indeed be more tricky (example: depending on parameters).

```
export default (options, { interval = 10, useClient = false } = {}) => {
  const Psql = useClient ? Client : Pool;
  const pool = new Psql(options);
  pool.connect = pool.connect.bind(pool);
  let connected = false;

  const connectionCheck = () => {
    setInterval(async () => {
      let client;

      try {
        client = await pool.connect();
        await client.query('SELECT 1');

        if (connected) return;
        connected = true;
        pool.emit('connect');
      } catch (err) {
        if (!connected) return;
        connected = false;
        pool.emit('disconnect', err);
      } finally {
        if (client) client.release();
      }
    }, interval * 1000);
  };

  pool.on('connect', () => {
    connected = true;
  });

  pool.on('error', (err) => {
    connected = false;
    pool.emit('disconnect', err);
  });

  pool.init = async () => connectionCheck();

  return pool;
};
```

**Comment by brunoluiz** [comment]
I discussed here with some people, and it seems to be more around handling `connect` events and informing if the connection with PG/Cockroach is degraded, so it can be used by things like Kubernetes or docker-compose. But, we decided that actually there might be other ways of doing this and, when this happens, is probably because our DB is having some trouble (other alerts will fire).

I am closing the issue now as it doesn't seem to be relevant.
